code/controllers/subsystems/goals.dm

```diff
@@ -16,34 +16,17 @@ SUBSYSTEM_DEF(goals)
 		/datum/goal/clean,
 		/datum/goal/money
 	)
-	var/list/departments = list()
 	var/list/ambitions =   list()
-
-/datum/controller/subsystem/goals/Initialize()
-	var/list/all_depts = subtypesof(/datum/department)
-	//See if map is very particular about what depts it has
-	if(LAZYLEN(GLOB.using_map.departments))
```

Comment:
Did you plan to readd the map-level department list at some point? You will need to remove the list from the map datum definition if not.

Reply:
Hmm not sure. I'll think about it. It might be relevant for end of round goals. If we add goals for security then that goal might show up despite there being jobs for security. So probably best to cull the department list. It might be nice to code it in a way to do it based of allowed jobs though.

Reply:
I've decided not to bother. Goals are the only area that hasn't been updated to check for what departments are in by job so I'll just add a check their. It's extra work for almost no payoff, and given I'm unlikely to clean up EVERY hard coded reference it's probably better for stability if all the unused default departments potter along noticed. I'll remove the list from the map datum definition.